pkg/server/server.go

```diff
@@ -143,6 +146,40 @@ func New(ctx context.Context, config *lepconfig.Config, endpoint string, cliUID
 		return nil, fmt.Errorf("failed to register sqlite metrics: %w", err)
 	}
 
+	scraper, err := scrape.NewScraper(ctx, promReg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create scraper: %w", err)
+	}
+	metricsStore, err := pkgmetricssqlite.NewStore(ctx, dbRW, dbRO, pkgmetricssqlite.DefaultTableName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create metrics store: %w", err)
+	}
+	go func() {
+		ticker := time.NewTicker(time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+			}
+
+			log.Logger.Infow("scraping metrics")
+			ms, err := scraper.Scrape(ctx)
+			if err != nil {
+				log.Logger.Errorw("failed to scrape metrics", "error", err)
+				continue
+			}
+
+			for _, m := range ms {
+				err := metricsStore.Record(ctx, m)
```

Comment:
If possible, we should support recording metrics in batch to obtain better performance.